teps/0078-tekton-catalog-support-tiers.md

```diff
@@ -0,0 +1,345 @@
+- Filter resources in the Tekton Hub based on support tiers.
+- Search resources with the Tekton CLI for resources in certain support tiers.
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accomodated.
+-->
+
+- Contributor should be able to indicate the support tier of a resource when authoring it.
+- Support tier information in the resource yaml in the Catalog should be optional.
+- Definitions and expectations of support tiers should be documented in the Tekton Catalog.
+- Available support tiers should be shown on the Tekton Hub, and users should be able to filter resources based on them.
+- Users should be able to search for resources of specific support tiers using the Tekton CLI.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Tekton Catalog
+
+Add a new `tekton.dev/tier` annotation to the resource yaml file with the relevant support tier, for example:
+
+```yaml
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/tier: verified
+```
+
+The `tekton.dev/tier` annotation should be optional. If the annotation is not specified, `community` support tier is 
+meant by default, as this support tier is actually default one in the Tekton Catalog. This would also provide backward 
+compatibility to the old versions of the resources in the Tekton Catalog.
+
+Information about support tier also should be added to the resource's `README.md` as separate `Support Tier` section to 
+establish quality expectations for the resource. The criteria to change the support tiers of resources (automated 
+testing, image scanning, ownership) should be documented in Tekton Catalog documentation for transparency to users and 
+contributors.
+
+### Tekton Hub
+
+Implement logic to fetch, process, store and update information about a resource's tier. The approach is similar to 
+what is available for `tags` and `platforms`. If there is no information about tier in the resource definition, 
+then the default `community` tier should be used.
+
+Hub UI should have `tier` filter and show resources based on it.
+
+CLI code should be extended:
+- to have `--tier` search parameter to filter resources based on support tiers
+- to have support tier information in the resource information view
+- to have support tier informatiion in the resource table
+
+### Tekton CLI
+
+After implementation in Tekton Hub, functionality can be extended to the tkn cli. Support tier information should be 
```

Comment:
just a imp nit, but it's actually the tkn-hub cli that implement that, tkn-cli simply include the cli into its own binary

Reply:
aah I see, still new to these projects so still figuring it out -- thanks 😀